pkg/ddevapp/craftcms.go

```diff
@@ -0,0 +1,170 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/drud/ddev/pkg/archive"
+	"github.com/drud/ddev/pkg/fileutil"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/pkg/errors"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// isCraftCmsApp returns true if the app is of type craftcms
+func isCraftCmsApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, "craft"))
+}
+
+// Set the Docroot to web
+func craftCmsConfigOverrideAction(app *DdevApp) error {
+	if app.Docroot == "" {
+		app.Docroot = "web"
+	}
+
+	return nil
+}
+
+// Returns the upload directory for importing files, if not already set
+func getCraftCmsUploadDir(app *DdevApp) string {
+	app.UploadDir = "files"
+
+	return app.UploadDir
+}
+
+// craftCmsImportFilesAction defines the workflow for importing project files.
+func craftCmsImportFilesAction(app *DdevApp, importPath, extPath string) error {
+	if app.UploadDir == "" {
+		return errors.Errorf("No upload_dir is set for this (craftcms) project")
+	}
+	destPath := app.GetHostUploadDirFullPath()
+
+	// parent of destination dir should exist
+	if !fileutil.FileExists(filepath.Dir(destPath)) {
+		return fmt.Errorf("unable to import to %s: parent directory does not exist", destPath)
```

Comment:
Do you think it's too dangerous to just create the directory? DDEV creates some settings directories if they don't exist, and it hasn't been a problem.

Reply:
This is a copy & paste from the current generic PHP setup so I'm assuming it's okay? If not, the generic PHP project, and others that have copy/pasted that code block should probably be changed too